[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47415 [quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu**

Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

jerryzh168 committed:
[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu

Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on test/quantization/test_quantize.py:
```diff
@@ -1295,6 +1298,25 @@ def checkNoFQModule(m):
+
+        m = M().train()
+        m.qconfig = default_qconfig
+        m = prepare_qat(m)
```
For ReLU6, would the same logic apply?

jerryzh168 replied:
yes, but relu6 requires some more changes, we need to merge the implementation of qunatized relu6 to the default one.

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]

z-a-f commented on docs/source/torch.nn.quantized.rst:
```diff
@@ -2,13 +2,12 @@ torch.nn.quantized
 
 This module implements the quantized versions of the nn layers such as
-~`torch.nn.Conv2d` and `torch.nn.ReLU`.
+~`torch.nn.Conv2d` and `torch.nn.Linear`.
```
I think at this point we can remove the "such as" examples.

jerryzh168 committed:
Update on "[quant] Remove nn.quantized.ReLU module and nn.quantized.functional.relu"


Summary:
nn.ReLU works for both float and quantized input, we don't want to define an nn.quantized.ReLU
that does the same thing as nn.ReLU, similarly for nn.quantized.functional.relu

this also removes the numerical inconsistency for models quantizes nn.ReLU independently in qat mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24747035](https://our.internmc.facebook.com/intern/diff/D24747035)

[ghstack-poisoned]
